ext/calendar/dow.c

```diff
@@ -35,11 +35,11 @@ int DayOfWeek(
 {
 	int dow;
 
-	dow = (sdn + 1) % 7;
+	dow = (int)(sdn % 7 + 1);
 	if (dow >= 0) {
-		return (dow);
+		return (dow % 7);
 	} else {
-		return (dow + 7);
+		return (dow + 7) % 7;
 	}
```

Comment:
Can't we just do:
````c
return (sdn % 7 + 8) % 7;
````
instead of what we have now?

Reply:
giving it a try

Reply:
I mean, it is somewhat obscure, and likely not what @nielsdos had in mind. But I think it should work.

Reply:
it does. I think the changed old test, the values make more sense to me.

Reply:
Without that `(int)` cast, bug_53574_2.phpt succeeds for me; either remove the cast, or cast the whole return value to `(int)` (that shouldn't be necessary since the `% 7` ensures that values fit into `int`, but if some tool complains, we may need to cast).

Reply:
Double modulus works and is intuitive as well. It should be possible to avoid the double modulus too but it likely doesn't matter for performance.